src/PartyDropper.sol

```diff
@@ -0,0 +1,120 @@
+
+contract PartyDropper is ERC1155Supply, Ownable {
+    struct Edition {
+        IPartyBid party;
+        string name;
+        string imageURI;
+        string description;
+    }
+    mapping(uint256 => Edition) public editions;
+    mapping(uint256 => mapping(address => bool)) public editionMinter;
+    mapping(uint256 => address) public editionCreator;
+    mapping(address => bool) public allowedCreators;
+    uint256 lastEditionId;
+
+    event EditionCreated(
+        uint256 indexed editionId,
+        address indexed creator,
+        address party,
+        string name
+    );
+    event Claimed(address indexed user, uint256 editionId);
+
+    modifier onlyAllowedCreator() {
+        require(
+            allowedCreators[msg.sender] || msg.sender == owner(),
+            "must be allowed creator or owner"
+        );
+        _;
+    }
+
+    constructor() ERC1155("") {}
+
+    // public functions
+    function mintEdition(uint256 editionId) public {
+        Edition storage fnd = editions[editionId];
+        require(
+            fnd.party.totalContributed(msg.sender) > 0,
+            "didn't contribute to PartyBid"
+        );
+        require(!editionMinter[editionId][msg.sender], "already minted");
+        editionMinter[editionId][msg.sender] = true;
+        _mint(msg.sender, editionId, 1, "");
+        emit Claimed(msg.sender, editionId);
+    }
+
+    function editionInfo(uint256 editionId)
+        public
+        view
+        returns (
+            address,
+            string memory,
+            string memory,
+            string memory
+        )
+    {
+        Edition storage fnd = editions[editionId];
+        return (address(fnd.party), fnd.name, fnd.imageURI, fnd.description);
+    }
+
+    function uri(uint256 editionId)
+        public
+        view
+        override
+        returns (string memory)
+    {
+        require(editionId <= lastEditionId, "must be created");
+        Edition storage edition = editions[editionId];
+        string memory json = Base64.encode(
+            bytes(
+                string(
+                    abi.encodePacked(
+                        '{"name": "',
+                        edition.name,
+                        '", "description": "',
+                        edition.description,
+                        '", "image": "',
+                        edition.imageURI,
+                        '"}'
+                    )
+                )
+            )
+        );
+        string memory output = string(
+            abi.encodePacked("data:application/json;base64,", json)
+        );
+        return output;
+    }
+
+    // allowedCreator functions
+    function createEdition(
+        address _party,
+        string memory _name,
+        string memory _imageURI,
+        string memory _description
+    ) public onlyAllowedCreator {
+        lastEditionId += 1;
+        IPartyBid party = IPartyBid(_party);
+        editions[lastEditionId] = Edition({
```

Comment:
Should we prevent creating multiple partydrops from the same party?  If we use the party address as the index instead of an edition counter we could both store on gas (don't save the address inside the struct, dont use lastEditionId) and be able to tell if a party has a collection already

```
editions[party] = Edition({
  name: _name,
  imageURI: _imageURI,
  description: _description
})
```

I guess gas isn't as important for this call, but it would prevent a possible nasty case of duplicate party favors.

Reply:
But i guess then we'd lose the ability to enumerate all party-favors.  Maybe we just want to prevent duplicates vs changing the counter.

Reply:
i appreciate this thinking, but my thinking is:

- i worry about gas for `mintFromEdition` but not so much for this`createEdition` fn
- i think there may be an actual use case for having 2 diff party favors for the same party